events/controller/status.go

```diff
@@ -450,3 +453,71 @@ func (p *natsEventStatusAcknowleger) nak() {
 
 	p.logger.Trace("event nak successful")
 }
+
+// HTTPConditionStatusPublisher implements the StatusPublisher interface to publish condition status updates over HTTP to NATS.
+type HTTPConditionStatusPublisher struct {
+	logger        *logrus.Logger
+	orcQueryor    orc.Queryor
+	conditionKind condition.Kind
+	conditionID   uuid.UUID
+	serverID      uuid.UUID
+	controllerID  registry.ControllerID
+}
+
+func NewHTTPConditionStatusPublisher(
+	serverID,
+	conditionID uuid.UUID,
+	conditionKind condition.Kind,
+	controllerID registry.ControllerID,
+	orcQueryor orc.Queryor,
+	logger *logrus.Logger,
+) ConditionStatusPublisher {
+	return &HTTPConditionStatusPublisher{
+		orcQueryor:    orcQueryor,
+		conditionID:   conditionID,
+		conditionKind: conditionKind,
+		serverID:      serverID,
+		controllerID:  controllerID,
+		logger:        logger,
+	}
+}
+
+func (s *HTTPConditionStatusPublisher) Publish(ctx context.Context, serverID string, state condition.State, status json.RawMessage, tsUpdateOnly bool) error {
+	_, span := otel.Tracer(pkgName).Start(
+		ctx,
+		"controller.status_http.Publish",
+		trace.WithSpanKind(trace.SpanKindConsumer),
+	)
+	defer span.End()
+
+	sv := &condition.StatusValue{
+		WorkerID:  s.controllerID.String(),
+		Target:    serverID,
+		TraceID:   trace.SpanFromContext(ctx).SpanContext().TraceID().String(),
+		SpanID:    trace.SpanFromContext(ctx).SpanContext().SpanID().String(),
+		State:     string(state),
+		Status:    status,
+		UpdatedAt: time.Now(),
+	}
+
+	resp, err := s.orcQueryor.ConditionStatusUpdate(ctx, s.conditionKind, s.serverID, s.conditionID, s.controllerID, sv, tsUpdateOnly)
+	if err != nil {
+		s.logger.WithError(err).Error("condition status update error")
+		return err
+	}
+
+	if resp.StatusCode != 200 {
+		// nolint:goerr113 // I'd like to keep the error definition close to where its in use.
+		errNon200 := fmt.Errorf("non 200 response code returned: %d", resp.StatusCode)
```

Comment:
This is pretty restrictive. A `201` would trip an error?

Reply:
As of now the API only returns a 200, we cover additional cases as they show up